core/Azure.Mcp.Core/src/Areas/Server/Commands/ToolLoading/RegistryToolLoader.cs

```diff
@@ -132,8 +138,11 @@ public override async ValueTask<CallToolResult> CallToolHandler(RequestContext<C
             };
         }
 
+        // For MCP servers loaded from registry.json, the ToolArea is also its "server name".
+        Activity.Current?.SetTag(TagName.ToolArea, kvp.Item1);
+
         var parameters = TransformArgumentsToDictionary(request.Params.Arguments);
-        return await mcpClient.CallToolAsync(request.Params.Name, parameters, cancellationToken: cancellationToken);
+        return await kvp.Item2.CallToolAsync(request.Params.Name, parameters, cancellationToken: cancellationToken);
```

Comment:
If you go with named tuple:

```suggestion
        return await kvp.Client.CallToolAsync(request.Params.Name, parameters, cancellationToken: cancellationToken);
```